base/strings/string.jl

```diff
@@ -328,6 +330,10 @@ isvalid(s::String, i::Int) = checkbounds(Bool, s, i) && thisind(s, i) == i
 
 isascii(s::String) = isascii(codeunits(s))
 
+# don't assume effects for general integers since we cannot know their implementation
+@assume_effects :foldable function repeat(c::Char, r::BitInteger)
+    @invoke repeat(c, r::Integer)
```

Comment:
Make sure to annotate argument type when you use `@invoke` when possible -- otherwise it will generate `Core.Typeof(c)` call, which can be a source of performance problem. This time is fine, since the type of `c` is known very concretely, but generally it's better to annotate the argument types for `@invoke` macro.